lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -331,22 +334,79 @@ private JourneyResponse getJourneyResponse(
+                // use pending return vcs to determine identity if available
+                if (!isNullOrEmpty(pendingReturnVcs)) {
+                    return new VerifiableCredentialBundle(pendingReturnVcs, true, true);
+                }
+                var currentVcs = vcs.get(CURRENT);
+                if (!isNullOrEmpty(currentVcs)) {
+                    return new VerifiableCredentialBundle(currentVcs, true, false);
+                }
             }
         }
-        return new VerifiableCredentialBundle(
-                verifiableCredentialService.getVcs(userId), false, false);
+        return new VerifiableCredentialBundle(tacticalVcs, false, false);
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    private void logIdentityMismatches(
+            List<VerifiableCredential> tacticalVcs,
+            Map<EvcsVCState, List<VerifiableCredential>> evcsEvcs) {
+
+        var migratedTacticalVcStrings =
+                tacticalVcs.stream()
+                        .filter(vc -> vc.getMigrated() != null)
+                        .map(VerifiableCredential::getVcString)
+                        .collect(Collectors.toSet());
+
+        // if we have pending vcs just check those
+        var evcsToCheck = evcsEvcs.get(PENDING);
+        if (isNullOrEmpty(evcsToCheck)) {
+            evcsToCheck = evcsEvcs.get(CURRENT);
+        }
+        var allTacticalVcStrings =
+                tacticalVcs.stream()
+                        .map(VerifiableCredential::getVcString)
+                        .collect(Collectors.toSet());
+        var evcsVcStrings =
+                isNullOrEmpty(evcsToCheck)
+                        ? new ArrayList<>()
+                        : evcsToCheck.stream()
+                                .map(VerifiableCredential::getVcString)
+                                .collect(Collectors.toSet());
+        var allEvcsVcStrings =
+                evcsEvcs.values().stream()
+                        .flatMap(List<VerifiableCredential>::stream)
+                        .map(VerifiableCredential::getVcString)
+                        .collect(Collectors.toSet());
+        var hasUnmigratedVcs = allTacticalVcStrings.size() > migratedTacticalVcStrings.size();
+
+        // check if we have unmigrated credentials alongside migrated ones
+        if (hasUnmigratedVcs && !migratedTacticalVcStrings.isEmpty()) {
+            LOGGER.warn("Unmigrated tactical credentials found alongside migrated credentials");
+        }
+
+        // check all the tactical vcs are in the selected evcs vcs
+        if (!hasUnmigratedVcs && !evcsVcStrings.containsAll(migratedTacticalVcStrings)) {
+            LOGGER.warn(
+                    "Failed to find corresponding evcs credential for migrated tactical credential");
+        }
+
+        // check all the evcs vcs are in the tactical store
+        if (!hasUnmigratedVcs && !migratedTacticalVcStrings.containsAll(allEvcsVcStrings)) {
```

Comment:
I think this check should just use `evcsVcStrings` as well - the tactical VCs should all be in the correct status in EVCS (i.e. pending or current, as appropriate)?